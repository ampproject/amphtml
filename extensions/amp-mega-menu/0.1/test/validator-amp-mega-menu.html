<!--
  Copyright 2019 The AMP HTML Authors. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS-IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the license.
-->
<!--
  Test Description:
  Tests for the amp-mega-menu tag.
-->
<!doctype html>
<html âš¡>
<head>
  <meta charset="utf-8">
  <link rel="canonical" href="amps.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-mega-menu" src="https://cdn.ampproject.org/v0/amp-mega-menu-0.1.js"></script>
  <script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
</head>
<body>
  <!-- Valid: amp-mega-menu with one expandable submenu and one link -->
  <amp-mega-menu layout="fixed-height" height="80">
    <nav>
      <ul>
        <li>
          <button>item</button>
          <div role="dialog">
            <ul>
              <li><div role="menu"></div></li>
            </ul>
          </div>
        </li>
        <li>
          <a>link</a>
        </li>
      </ul>
    </nav>
  </amp-mega-menu>
  <!-- Valid: amp-mega-menu with dynamic template rendering -->
  <amp-mega-menu layout="fixed-height" height="80">
    <amp-list layout="fill" src=".">
      <template type="amp-mustache">
        <nav>
          <ul>
            {{items}}
            <li>
              <div role="button">{{name}}</div>
              <div role="dialog"></div>
            </li>
            {{items}}
          </ul>
        </nav>
      </template>
    </amp-list>
  </amp-mega-menu>
</body>
</html>
