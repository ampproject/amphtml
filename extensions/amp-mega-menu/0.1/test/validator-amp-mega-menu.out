FAIL
|  <!--
|    Copyright 2019 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    Tests for the amp-mega-menu tag.
|  -->
|  <!doctype html>
|  <html âš¡>
|  <head>
|    <meta charset="utf-8">
|    <link rel="canonical" href="amps.html">
|    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|    <script async src="https://cdn.ampproject.org/v0.js"></script>
|    <script async custom-element="amp-mega-menu" src="https://cdn.ampproject.org/v0/amp-mega-menu-0.1.js"></script>
|    <script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
|    <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
|  </head>
|  <body>
|    <!-- Valid: amp-mega-menu with one expandable submenu and one link -->
|    <amp-mega-menu layout="fixed-height" height="80">
|      <nav>
|        <ul>
|          <li>
|            <button>item</button>
|            <div role="dialog">
|              <ul>
|                <li><div role="menu"></div></li>
|              </ul>
|            </div>
|          </li>
|          <li>
|            <a>link</a>
|          </li>
|        </ul>
|      </nav>
|    </amp-mega-menu>
|    <!-- Valid: amp-mega-menu with dynamic template rendering -->
|    <amp-mega-menu layout="fixed-height" height="80">
>>   ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:52:2 The tag 'amp-mega-menu' appears more than once in the document. (see https://amp.dev/documentation/components/amp-mega-menu)
|      <amp-list layout="fill" src=".">
|        <template type="amp-mustache">
|          <nav>
|            <ul>
|              {{items}}
|              <li>
|                <div role="button">{{name}}</div>
|                <div role="dialog"></div>
|              </li>
|              {{items}}
|            </ul>
|          </nav>
|        </template>
|      </amp-list>
|    </amp-mega-menu>
|    <!-- Invalid: amp-mega-menu with incorrect DOM -->
|    <amp-mega-menu layout="fixed-height" height="80">
>>   ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:69:2 The tag 'amp-mega-menu' appears more than once in the document. (see https://amp.dev/documentation/components/amp-mega-menu)
|      <nav>
>>     ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:70:4 Tag 'AMP-MEGA-MENU > NAV' must have 1 child tags - saw 2 child tags.
|        <!-- nav cannot contain more than one child, which must be ul/ol -->
|        <div></div>
>>       ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:72:6 Tag 'div' is disallowed as child of tag 'AMP-MEGA-MENU > NAV'. Child tag must be one of ['ol', 'ul'].
|        <ul>
|          <!-- ul/ol cannot have elements other than li as children -->
|          <div></div>
>>         ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:75:8 Tag 'div' is disallowed as child of tag 'AMP-MEGA-MENU NAV > UL/OL'. Child tag must be one of ['li'].
|          <!-- menu items must have at least one child -->
|          <li></li>
>>         ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:77:8 Tag 'AMP-MEGA-MENU NAV > UL/OL > LI' must have a minimum of 1 child tags - saw 0 child tags.
>>         ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:77:8 The mandatory reference point 'AMP-MEGA-MENU item-heading' for 'AMP-MEGA-MENU NAV > UL/OL > LI' is missing.
|          <!-- menu item cannot have multiple children unless one has role="dialog" -->
|          <li>
>>         ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:79:8 The reference point 'AMP-MEGA-MENU item-heading' for 'AMP-MEGA-MENU NAV > UL/OL > LI' must be unique but a duplicate was encountered.
|            <button></button>
|            <button></button>
|          </li>
|          <!-- role="dialog" must be accompanied by a sibling that toggles it -->
|          <li>
>>         ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:84:8 The mandatory reference point 'AMP-MEGA-MENU item-heading' for 'AMP-MEGA-MENU NAV > UL/OL > LI' is missing.
|            <div role="dialog"></div>
|          </li>
|          <!-- menu item cannot have multiple role="dialog" as children -->
|          <li>
>>         ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:88:8 The reference point 'AMP-MEGA-MENU item-content' for 'AMP-MEGA-MENU NAV > UL/OL > LI' must be unique but a duplicate was encountered.
|            <button></button>
|            <div role="dialog"></div>
|            <div role="dialog"></div>
|          </li>
|          <!-- role=menu is disallowed as child of menu item -->
|          <li>
|            <button></button>
|            <div role="menu"></div>
>>           ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:96:10 The tag 'div', a child tag of 'AMP-MEGA-MENU NAV > UL/OL > LI', does not satisfy one of the acceptable reference points: AMP-MEGA-MENU item-content, AMP-MEGA-MENU item-heading.
|          </li>
|          <!-- there cannot be a nav element inside another nav -->
|          <li>
|            <button></button>
|            <div role="dialog">
|              <nav></nav>
>>             ^~~~~~~~~
amp-mega-menu/0.1/test/validator-amp-mega-menu.html:102:12 The parent tag of tag 'AMP-MEGA-MENU > NAV' is 'div', but it can only be 'amp-mega-menu'.
|            </div>
|          </li>
|        </ul>
|      </nav>
|    </amp-mega-menu>
|  </body>
|  </html>